src/Microsoft.NetCore.Analyzers/Core/Runtime/DisposeMethodsShouldCallBaseClassDispose.cs

```diff
@@ -36,8 +36,7 @@ public sealed class DisposeMethodsShouldCallBaseClassDispose : DiagnosticAnalyze
                                                                              helpLinkUri: "https://docs.microsoft.com/visualstudio/code-quality/ca2215-dispose-methods-should-call-base-class-dispose",
                                                                              customTags: FxCopWellKnownDiagnosticTags.PortedFxCopRule);
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray<DiagnosticDescriptor>.Empty;
-        //ImmutableArray.Create(Rule);
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
```

Comment:
@mavasani Is there a reason this analyzer was disabled by removing diagnostic IDs?

Reply:
ping @mavasani

Reply:
@mavasani Do you have any info regarding why the rule was disabled?

Reply:
I can't recall as of now, but I would rather we not do this change in this PR. Can we file a separate issue to track this?

Reply:
https://github.com/dotnet/roslyn-analyzers/issues/3141

Reply:
@mavasani @sharwell I cannot undo this change otherwise the tests are failing because the issue isn't raised.

Reply:
IMO that seems like a drawback/regression from the testing library that should be fixed. It shouldn't be requiring non-empty supported diagnostics just to enable testing of the analyzer.

Reply:
@mavasani The old test framework doesn't support this case either. It just silently passes all the tests by returning immediately here:

https://github.com/dotnet/roslyn-analyzers/blob/8e719bd21d008691303e385a76f6ec1a372ed51b/src/Test.Utilities/DiagnosticAnalyzerTests.Extensions.cs#L32-L36

Reply:
Oh, that's even worse :-). I am fine with this change and we see any fallouts, we can handle it subsequently.